Updated layers.json

arcayngel opened the pull request:
Updated Layers file to include CAF (Canadian Armed Forces) DLC.

Thomas-Smyth commented on layers.json:
```diff
@@ -1177,6 +1240,15 @@
     "minPlayers": 65,
     "maxPlayers": 80
+  },
+    "Yehorivka CAF RAAS v1": {
```
Can we double-check this name lacks underscores please?

The logs file seems to suggest it does:
```
[2019.08.06-22.35.20:320][  0]LogSquadTrace: [DedicatedServer]USQGameInstance::AddModMapsToMapList(): Added Modded Map: Yehorivka_CAF_RAAS_v1
```

arcayngel replied:
Fixed in latest patch. Map lacked underscores. This map naming is driving me nuts with its lack of standardization...

Thomas-Smyth commented on layers.json:
```diff
@@ -367,6 +367,15 @@
+    "Gorodok_Invasion_v2_CAF": {
+    "map": "Gorodok",
+    "gamemode": "Invasion",
+    "version": "v2 CAF",
```
Not too sure about this version. Rather frustrating the lack of consistent naming in v15.3... I do rather think nothing went through QA.

I imagine this is going to be removed at some point or at least the map naming made consistent.

I suggest we add a new field called `dlc` and remove `CAF` from the version (though not from the name or that will cause matching issues). For example:
```
"Manic_RAAS_v1": {
    "map": "Manic",
    "gamemode": "RAAS",
    "version": "v1",
    "night": false,
    "middleFlags": [],
    "minPlayers": 65,
    "maxPlayers": 80
    "dlc": "CanadianArmedForces"
  },
```

Thomas-Smyth replied:
Note - I am using the name `CanadianArmedForces` due to that being used in the logs.

```
[2019.08.06-22.35.20:319][  0]LogModdingRuntimeModule: Display: Successfully mounted and serialised mod: CanadianArmedForces
```

Whether that is some identifier for the mod I do not know, but I think it's the best to go by for now.

arcayngel replied:
This can be done. If we implement the "dlc" field then I can go back through and add that field. Note that maps such as Manic and Nanisivik do not contain "_CAF" in their name as they're a brand new map, unlike say Yeho or Goro.

Thomas-Smyth commented on layers.json:
```diff
@@ -367,6 +367,15 @@
     "minPlayers": 65,
     "maxPlayers": 80
+  },
+    "Gorodok_Invasion_v2_CAF": {
```
Can you please unindent the name of the layer so that it is inline with the `},`? The rest of the layer object is ok 👌 

I think I used [this site](https://jsonformatter.org/) in the past, but if Git picks up too many changes just do it manually please.

arcayngel replied:
Changes implemented as of latest patch-1

Thomas-Smyth commented on layers.json:
```diff
@@ -727,6 +736,33 @@
     "minPlayers": 0,
     "maxPlayers": 36
+  },
+    "Manic_RAAS_v1": {
```
Same again with the indenting for this and all the other layers you added.

arcayngel replied:
Changes implemented as of latest patch-1

Thomas-Smyth commented on layers.json:
```diff
@@ -727,6 +736,33 @@
+    "gamemode": "Skirmish",
+    "version": "v1",
+    "night": false,
+    "middleFlags": [],
```
Does this have no middle flag(s)?

Mind finding out for me or letting me know please and I'll test it on an empty server at some point.

arcayngel replied:
It does have a middle flag, however I was unable to confirm before pushing the code. Squad Maps does not have the layer up so this will have to be confirmed.

arcayngel commented:
> The logs suggest we might be missing some more layers. Mind checking for me, please? I assume they are mostly broken / non existant layers.
> 
> ```
> [2019.08.06-22.35.20:319][  0]LogSquadTrace: [DedicatedServer]USQGameInstance::AddModMapsToMapList(): Added Modded Map: LL_Daytime
> [2019.08.06-22.35.20:319][  0]LogSquadTrace: [DedicatedServer]USQGameInstance::AddModMapsToMapList(): Added Modded Map: LL_SnowStorm
> [2019.08.06-22.35.20:320][  0]LogSquadTrace: [DedicatedServer]USQGameInstance::AddModMapsToMapList(): Added Modded Map: Manic-5
> [2019.08.06-22.35.20:320][  0]LogSquadTrace: [DedicatedServer]USQGameInstance::AddModMapsToMapList(): Added Modded Map: Nanisivik_Overcast
> [2019.08.06-22.35.20:320][  0]LogSquadTrace: [DedicatedServer]USQGameInstance::AddModMapsToMapList(): Added Modded Map: Kohat_Invasion_v2
> [2019.08.06-22.35.20:320][  0]LogSquadTrace: [DedicatedServer]USQGameInstance::AddModMapsToMapList(): Added Modded Map: Kohat_Invasion_v3
> [2019.08.06-22.35.20:320][  0]LogSquadTrace: [DedicatedServer]USQGameInstance::AddModMapsToMapList(): Added Modded Map: Kohat_RAAS_v4
> [2019.08.06-22.35.20:320][  0]LogSquadTrace: [DedicatedServer]USQGameInstance::AddModMapsToMapList(): Added Modded Map: LL_Manic_FoggyMorning
> ```
> 
> It is really frustrating the lack of consistency and reliability in this game. For the last few updates it feels like QA has been none existant. I'm sure it cannot be hard to rename a layer, so why don't they have this sorted _*sigh*_
> 
> Once these issues are resolved I will accept the PR. It's not perfect, but that's not our fault it's OWI's for the reasons above. Hopefully, when they sort themselves out for v15.4/v16 we can tidy up.
> 
> If not then I think we need to restructure out approach to what layers we add. In particular, adding everything in the logs and having a `broken` field if we know it does not work and not adding the details for any layers we cannot load to find what they are.
> 
> If you message me on Discord when you are free and I'm around I'll set you up with write access on this repo. It will be easier that way as I can make the minor changes to the branch rather than making a code review like this.
> 
> **Thanks a lot for doing this!**

Kohat Invasion v2 is included in updated json. If its still popping up will have to look at it more. 
Kohat Invasion v3 no longer exists within the game looking at the wiki, so it isn't included.
Kohat RAAS v4 added as of latest patch-1 push.
All other maps in this list are testing only and should not appear in the applet at all. (Nanisivik Overcast for example is a lighting layer and shouldn't be in the game at all let alone on a map rotation).
